internal/jobservice/repository/sql_job_service.go

```diff
@@ -0,0 +1,266 @@
+package repository
+
+import (
+	"database/sql"
+	"fmt"
+	"sync"
+	"time"
+
+	_ "modernc.org/sqlite"
+
+	"github.com/G-Research/armada/internal/jobservice/configuration"
+	js "github.com/G-Research/armada/pkg/api/jobservice"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// Internal structure for storing in memory JobTables and Subscription JobSets
+// Locks are used for concurrent access of map
+type JobStatus struct {
+	jobMap        map[string]*JobTable
+	jobLock       sync.RWMutex
+	subscribeMap  map[string]*SubscribeTable
+	subscribeLock sync.RWMutex
+}
+
+func NewJobStatus(jobMap map[string]*JobTable, subscribeMap map[string]*SubscribeTable) *JobStatus {
+	return &JobStatus{jobMap: jobMap, subscribeMap: subscribeMap}
+}
+
+// SQLJobService for persisting to DB.
+type SQLJobService struct {
+	jobStatus        *JobStatus
+	jobServiceConfig *configuration.JobServiceConfiguration
+	db               *sql.DB
+}
+
+func NewSQLJobService(jobMap *JobStatus, config *configuration.JobServiceConfiguration, db *sql.DB) *SQLJobService {
+	return &SQLJobService{jobStatus: jobMap, jobServiceConfig: config, db: db}
+}
+
+type SubscribedTuple struct {
+	Queue  string
+	JobSet string
+}
+
+// Create a Table from a hard-coded schema.
+// Open to suggestions on how to make this better
+func (s *SQLJobService) CreateTable() {
+	_, err := s.db.Exec("DROP TABLE IF EXISTS jobservice")
+	if err != nil {
+		panic(err)
+	}
+	_, err = s.db.Exec(`
+CREATE TABLE jobservice (
+Queue TEXT,
+JobSetId TEXT,
+JobId TEXT,
+JobResponseState TEXT,
+JobResponseError TEXT,
+Timestamp INT
+)`)
+	if err != nil {
+		panic(err)
+	}
+}
+
+// Get the JobStatus given the jodId
+func (s *SQLJobService) GetJobStatus(jobId string) (*js.JobServiceResponse, error) {
+	s.jobStatus.jobLock.RLock()
+	jobResponse, ok := s.jobStatus.jobMap[jobId]
+	s.jobStatus.jobLock.RUnlock()
+	if !ok {
+		sqlResponse, errSql := s.GetJobStatusSQL(jobId)
+		if sqlResponse != nil {
```

Comment:
won't this return an error if the job isn't present yet?  Is that correct?

Reply:
Actually I checked and it won't because  `GetJobStatusSQL` returns (`notFound`, `nil`) in this case.  I think you probably want to change `if sqlResponse != nil` to `if err != nil` on line 74